### PR TITLE
Add Table Wizard

### PR DESCRIPTION
**Pull request by khe0613**
Add Table Wizard!
#424 
<br/><br/><br/>

table wizard 4
First, click Edit on the menu bar and click the Table editing mode.
<br/><br/><br/><br/>

table wizard 1
Click Edit on the menu bar and click Table Wizard
<br/><br/><br/><br/>

table wizard 2
You can type the number of rows and columns for the table you want to create.
Click the OK button to create a table.
<br/><br/><br/><br/>

table wizard 3
Now, you can complete the table by simply typing in the contents of the table.

### CODE REVIEW

File: ReText/table.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+from ReText import globalSettings
+from PyQt5.QtCore import pyqtSignal, Qt
+from PyQt5.QtWidgets import QDialog, QVBoxLayout, QTabWidget, \
+	QDialogButtonBox, QWidget, QGridLayout, QLabel, QSpinBox
```
Comment (mitya57) [inline ReText/table.py]: In new files please use 4 spaces indentation. Old files have indentation by tabs but that is legacy and I want to remove that in the future.

File: ReText/table.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+class ClickableLabel(QLabel):
+	clicked = pyqtSignal()
```
Comment (mitya57) [inline ReText/table.py]: This signal is not actually used, so you can just use QLabel here.

File: ReText/table.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+			displayname, name = option[:2]
+			
+			if displayname:
+				label = ClickableLabel(displayname + ':', self)
+			
+			value = getattr(globalSettings, name)
+			self.configurators[name] = QSpinBox(self)
+			if (name == 'rowCount') or (name == 'columnCount'):
+				self.configurators[name].setRange(1,10)
+
```
Comment (mitya57) [inline ReText/table.py]: Copying “configurators” code from config sounds like overkill. You can just hard-code two labels and two inputs here.

File: ReText/__main__.py [hunk 4]
```diff
@@ -33,6 +33,7 @@
 from PyQt5.QtWidgets import QApplication
 from PyQt5.QtNetwork import QNetworkProxyFactory
 
+
```
[the remark this answers is not in the document]
Comment (khe0613) [reply ReText/__main__.py]: Yes, I remove it!

File: ReText/config.py [hunk 5]
```diff
@@ -151,6 +151,7 @@ def getPageWidget(self, options):
 				layout.addWidget(self.configurators[name], index, 0, 1, 2)
 				continue
 			value = getattr(globalSettings, name)
+
```
[the remark this answers is not in the document]
Comment (khe0613) [reply ReText/config.py]: Yes, I remove it!

File: ReText/table.py (hunk 1 above)
Comment (khe0613) [reply ReText/table.py]: I modified to use 4 spaces indentation instead of Tab!

File: ReText/table.py (hunk 2 above)
Comment (khe0613) [reply ReText/table.py]: OK, I remove the code for the line 8.

File: ReText/table.py (hunk 3 above)
Comment (khe0613) [reply ReText/table.py]: I modified the code according to your advice!

File: ReText/__init__.py [hunk 6]
```diff
@@ -100,6 +100,8 @@ def getBundledIcon(iconName):
 	'useWebKit': False,
 	'windowGeometry': QByteArray(),
 	'paperSize': '',
+	'rowCount': 3,
+	'columnCount':3
```
[the remark this answers is not in the document]
Comment (khe0613) [reply ReText/__init__.py]: So I removed this part because I don't need it anymore.
Thank you for your feedback.

**Comment by khe0613** [comment]
I modified code to reflect your feedback.
Thank you for your feedback.

**Comment by mitya57** [comment]
I have refactored the code and merged. Thanks!

**Pull request merged**